src/precice/bindings/python/setup.py

```diff
@@ -1,62 +1,58 @@
 import sys
 import os
 import shutil
+import subprocess
+from enum import Enum
 
 from distutils.core import setup
 from distutils.extension import Extension
 from Cython.Distutils import build_ext
 
-# check if PRECICE_ROOT is defined
-if not os.getenv('PRECICE_ROOT'):
-    print("ERROR: PRECICE_ROOT not defined!")
-    exit(1)
-
-precice_root = os.getenv('PRECICE_ROOT')
-python_bindings_path = os.path.join(precice_root,'src','precice','bindings','python')
+python_bindings_path = os.path.dirname(os.path.abspath(__file__))
+precice_root = os.path.join(python_bindings_path,"../../../../../precice")
```

Comment:
Why not $PRECICE_ROOT here?

It's ok, if you prefer the relative location, not sure if using PRECICE_ROOT is superior here.

Reply:
I used `$PRECICE_ROOT` before. But this gave me problems, when I had to run `sudo python setup.py install`, if the python is installed in a folder without user access.